src/super_gradients/training/metrics/obb_detection_metrics.py

```diff
@@ -0,0 +1,631 @@
+
+        greater_component_is_better = [
+            (self.precision_metric_key, True),
+            (self.recall_metric_key, True),
+            (self.map_metric_key, True),
+            (self.f1_metric_key, True),
+        ]
+
+        if self.include_classwise_ap:
+            self.per_class_ap_names = [f"{state_dict_prefix}AP{self._get_range_str()}_{class_name}" for class_name in class_names]
+            greater_component_is_better += [(key, True) for key in self.per_class_ap_names]
+
+        self.greater_component_is_better = collections.OrderedDict(greater_component_is_better)
+        self.component_names = list(self.greater_component_is_better.keys())
+        self.calc_best_score_thresholds = calc_best_score_thresholds
+        self.best_threshold_per_class_names = [f"Best_score_threshold_{class_name}" for class_name in class_names]
+
+        if self.calc_best_score_thresholds:
+            self.component_names.append("Best_score_threshold")
+
+        if self.calc_best_score_thresholds and self.include_classwise_ap:
+            self.component_names += self.best_threshold_per_class_names
+
+        self.components = len(self.component_names)
+
+        self.post_prediction_callback = post_prediction_callback
+        self.is_distributed = super_gradients.is_distributed()
+        self.world_size = None
+        self.rank = None
+        self.state_key = f"{state_dict_prefix}matching_info{self._get_range_str()}"
+        self.add_state(self.state_key, default=[], dist_reduce_fx=None)
+
+        self.recall_thresholds = torch.linspace(0, 1, 101) if recall_thres is None else torch.tensor(recall_thres, dtype=torch.float32)
+        self.score_threshold = score_thres
+        self.top_k_predictions = top_k_predictions
+
+        self.accumulate_on_cpu = accumulate_on_cpu
+
+    def update(self, preds, gt_samples: List[OBBSample]) -> None:
+        """
+        Apply NMS and match all the predictions and targets of a given batch, and update the metric state accordingly.
+
+        :param preds:           Raw output of the model, the format might change from one model to another,
+                                but has to fit the input format of the post_prediction_callback (cx,cy,wh)
+        :param target:          Targets for all images of shape (total_num_targets, 6) LABEL_CXCYWH. format:  (index, label, cx, cy, w, h)
+        :param device:          Device to run on
+        :param inputs:          Input image tensor of shape (batch_size, n_img, height, width)
+        :param crowd_targets:   Crowd targets for all images of shape (total_num_targets, 6), LABEL_CXCYWH
+        """
+        preds: List[OBBPredictions] = self.post_prediction_callback(preds)
+        output_device = "cpu" if self.accumulate_on_cpu else None
+        matching_strategy = OBBIoUMatching(self.iou_thresholds.to(preds[0].scores.device))
+
+        for pred, trues in zip(preds, gt_samples):
+            image_mathing = compute_obb_detection_matching(
```

Comment:
Isn't update the only different method between this implementation and the original DetectionMetrics?
Seems to me like an overkill to re-write the entire metric...
We can probably even shrink it even more (not gonna be annoying about this though) and extract the matching parts + iou claculations to some methods. But only if it isn't too much work.

Reply:
Fixed in 75a47e1761d8c7131436532ec0d39a9c2ea4f0fb